onnx_caffe2/backend.py

```diff
@@ -384,6 +387,70 @@ def _create_lstm(cls, init_model, pred_model, n, opset_version):
                          list(init_net.Proto().op),
                          list(pred_mh.Proto().external_input))
 
+    @classmethod
+    def _create_gru(cls, init_model, pred_model, n, opset_version):
+        assert init_model is not None, "cannot convert GRUs through this codepath"
+        assert pred_model is not None, "cannot convert GRUs through this codepath"
+
+        attrs = dict(n.attrs)
+        hidden_size = attrs.pop('hidden_size')
+        linear_before_reset = attrs.pop('linear_before_reset')
+        assert not attrs, "unsupported GRU attributes: " + str(attrs.keys())
+
+        input_blob, W, R, B, sequence_lens, initial_h = n.inputs
+
+        input_size = cls._rnn_shape_inference(init_model, pred_model, n, input_blob, W)
+        assert input_size is not None, "best-effort shape inference for GRU input failed"
```

Comment:
Python asserts can be compiled away, so it's best to do an explicit test and raise exception if user input could induce the error.